testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java

```diff
@@ -64,13 +66,59 @@ private void reportUnannotatedTestMethods(ITestClass testClass)
         List<Method> unannotatedTestMethods = findUnannotatedTestMethods(realClass);
         if (!unannotatedTestMethods.isEmpty()) {
             reportListenerFailure(
-                    ReportUnannotatedMethods.class,
+                    ReportBadTestAnnotations.class,
                     "Test class %s has methods which are public but not explicitly annotated. Are they missing @Test?%s",
                     realClass.getName(),
                     unannotatedTestMethods.stream()
                             .map(Method::toString)
                             .collect(joining("\n\t\t", "\n\t\t", "")));
         }
+
+        if (!realClass.isAnnotationPresent(Suppress.class)) {
+            Optional<Class<?>> clazz = classWithMeaninglessTestAnnotation(realClass);
+            if (clazz.isPresent()) {
+                reportListenerFailure(
+                        ReportBadTestAnnotations.class,
+                        "Test class %s (%s) has meaningless class-level @Test annotation. We require each test method be explicitly " +
+                                "annotated, deliberately not leveraging https://testng.org/doc/documentation-main.html#class-level.",
+                        clazz.get().getName(),
+                        realClass.getName());
+            }
+        }
+    }
+
+    @VisibleForTesting
+    static Optional<Class<?>> classWithMeaninglessTestAnnotation(Class<?> realClass)
+    {
+        for (Class<?> clazz = realClass; clazz != null; clazz = clazz.getSuperclass()) {
+            org.testng.annotations.Test testAnnotation = clazz.getAnnotation(org.testng.annotations.Test.class);
+            if (testAnnotation != null && isAllDefaults(testAnnotation, org.testng.annotations.Test.class)) {
+                return Optional.of(clazz);
+            }
+        }
+        return Optional.empty();
+    }
+
+    private static <T extends Annotation> boolean isAllDefaults(T annotationInstance, Class<T> annotationClass)
```

Comment:
`isAllDefaults` is always called with the same second argument. Could be removed, yeah?